hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -382,8 +382,6 @@ object DataSourceWriteOptions {
 
   // Avro Kafka Source configs
   val KAFKA_AVRO_VALUE_DESERIALIZER = "hoodie.deltastreamer.source.kafka.value.deserializer.class"
-
-  // Schema provider class to be set to be used in custom kakfa deserializer
-  val SCHEMA_PROVIDER_CLASS_PROP = "hoodie.deltastreamer.schemaprovider.class"
```

Comment:
@nsivabalan do we need a story for deprecating this first? seems like we just replaced A with B and it means existing jobs are going to break out-of-box?

Reply:
We did not replace A with B. we have a config that determines the schemaProvider class. unless someone sets this config value, behavior stays the same (old behavior).
https://github.com/apache/hudi/blob/b32855545bb9dbc6f49c18d1586585a39abe5004/hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/AvroKafkaSource.java#L60
check line 60 to 64.

Reply:
Both settings (old and new) are not set via config. Bot are calculated and (over)written by Deltastreamer itself. They are used to pass things around.

Reply:
okay makes sense. Having this with options is kind of confusing and I thought this is user configurable. 

While I have both of you here, are we turning this on by default now?

Reply:
I'm fine to define the config name in an other place (e.g. in KafkaOffsetGen). Should I change it with a new PR?
I would suggest turning it on by default, what are your thoughts @nsivabalan and @vburenin?

Reply:
Created https://github.com/apache/hudi/pull/3172 that adresses this